jpa/hibernate5_3/src/main/java/org/jboss/as/jpa/hibernate5/HibernatePersistenceProviderAdaptor.java

```diff
@@ -99,32 +99,62 @@ public void addProviderDependencies(PersistenceUnitMetadata pu) {
         final Properties properties = pu.getProperties();
         final String sharedCacheMode = properties.getProperty(SHARED_CACHE_MODE);
 
-        if ( Classification.NONE.equals(platform.defaultCacheClassification())) {
+        if (Classification.NONE.equals(platform.defaultCacheClassification())) {
             JPA_LOGGER.tracef("second level cache is not supported in platform, ignoring shared cache mode");
             pu.setSharedCacheMode(SharedCacheMode.NONE);
         }
-        // check if 2lc is explicitly disabled which takes precedence over other settings
-        boolean sharedCacheDisabled = SharedCacheMode.NONE.equals(pu.getSharedCacheMode())
-                ||
-                NONE.equals(sharedCacheMode) ||
-                SharedCacheMode.UNSPECIFIED.equals(pu.getSharedCacheMode()) ||
-                UNSPECIFIED.equals(sharedCacheMode);
-        if (!sharedCacheDisabled &&
-                (null == properties.getProperty(AvailableSettings.USE_SECOND_LEVEL_CACHE) ||
-                        Boolean.parseBoolean(properties.getProperty(AvailableSettings.USE_SECOND_LEVEL_CACHE))))
-                {
+        // precedence order of cache settings (1 overrides other settings and 3 is lowest precedence level):
+        // 1 - SharedCacheMode.NONE
+        // 2 - AvailableSettings.USE_SECOND_LEVEL_CACHE
+        // 2 - AvailableSettings.USE_QUERY_CACHE
+        // 3 - SharedCacheMode.UNSPECIFIED =
```

Comment:
Does that '=' belong there?  Trying to review this it made me think L111/112 were somehow continuations of L110 but I don't think that's the case.

Reply:
oops, good catch, removing...